infra/blueprint-test/pkg/bq/bq.go

```diff
@@ -0,0 +1,165 @@
+func newCmdConfig(opts ...cmdOption) (*CmdCfg, error) {
+	gOpts := &CmdCfg{}
+	// apply options
+	for _, opt := range opts {
+		opt(gOpts)
+	}
+	if gOpts.bqBinary == "" {
+		err := utils.BinaryInPath("bq")
+		if err != nil {
+			return nil, err
+		}
+		gOpts.bqBinary = "bq"
+	}
+	if gOpts.commonArgs == nil {
+		gOpts.commonArgs = []string{"--format", "json"}
+	}
+	if gOpts.logger == nil {
+		gOpts.logger = utils.GetLoggerFromT()
+	}
+	return gOpts, nil
+}
+
+// RunCmd executes a bq command and fails test if there are any errors.
+func RunCmd(t testing.TB, cmd string, opts ...cmdOption) string {
+	op, err := RunCmdE(t, cmd, opts...)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return op
+}
+
+// RunCmdE executes a bq command and return output.
+func RunCmdE(t testing.TB, cmd string, opts ...cmdOption) (string, error) {
+	gOpts, err := newCmdConfig(opts...)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// split command into args
+	args := strings.Fields(cmd)
+	bqCmd := shell.Command{
+		Command: "bq",
+		Args:    append(args, gOpts.commonArgs...),
+		Logger:  gOpts.logger,
+	}
+	return shell.RunCommandAndGetStdOutE(t, bqCmd)
+}
+
+// Run executes a bq command and returns value as gjson.Result.
+// It fails the test if there are any errors executing the bq command or parsing the output value.
+func Run(t testing.TB, cmd string, opts ...cmdOption) gjson.Result {
+	op := RunCmd(t, cmd, opts...)
+	if !gjson.Valid(op) {
+		t.Fatalf("Error parsing output, invalid json: %s", op)
+	}
+	return gjson.Parse(op)
+}
+
+// RunWithCmdOptsf executes a bq command and returns value as gjson.Result.
+//
+// RunWithCmdOptsf(t, ops.., "ls --datasets --project_id=%s", "projectId")
+//
+// It fails the test if there are any errors executing the bq command or parsing the output value.
+func RunWithCmdOptsf(t testing.TB, opts []cmdOption, cmd string, args ...interface{}) gjson.Result {
+	return Run(t, stringFromTextAndArgs(append([]interface{}{cmd}, args...)...), opts...)
+}
+
+// Runf executes a bq command and returns value as gjson.Result.
+//
+// Runf(t, "ls --datasets --project_id=%s", "projectId")
+//
+// It fails the test if there are any errors executing the bq command or parsing the output value.
+func Runf(t testing.TB, cmd string, args ...interface{}) gjson.Result {
+	return Run(t, stringFromTextAndArgs(append([]interface{}{cmd}, args...)...))
+}
+
+// stringFromTextAndArgs convert msg and args to formatted text
+func stringFromTextAndArgs(msgAndArgs ...interface{}) string {
+	if len(msgAndArgs) == 0 || msgAndArgs == nil {
+		return ""
+	}
+	if len(msgAndArgs) == 1 {
+		msg := msgAndArgs[0]
+		if msgAsStr, ok := msg.(string); ok {
+			return msgAsStr
+		}
+		return fmt.Sprintf("%+v", msg)
+	}
+	if len(msgAndArgs) > 1 {
+		return fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+	return ""
+}
+
+// ActivateCredsAndEnvVars activates credentials and exports auth related envvars.
+func ActivateCredsAndEnvVars(t testing.TB, creds string) {
```

Comment:
This does not seem BQ specific and might be a better fit in utils?

Reply:
I copied this directly from `gcloud.go` since auth is the same - I can move it from here (and there) to utils if that makes sense. Or I can leave it as-is in `gcloud.go`